src/System.ComponentModel.Annotations/tests/MaxLengthAttributeTests.cs

```diff
@@ -111,20 +112,13 @@ public static void GetValidationResult_ValueNotStringOrICollection_ThrowsInvalid
         [Fact]
         public static void GetValidationResult_ValueGenericICollection_ThrowsInvalidCastException()
         {
-            Assert.Throws<InvalidCastException>(() => new MaxLengthAttribute().GetValidationResult(new GenericICollectionClass(), new ValidationContext(new object())));
+            Assert.Throws<InvalidCastException>(() => new MaxLengthAttribute().GetValidationResult(new GenericIEnumerableClass(), new ValidationContext(new object())));
         }
     }
 
-    class GenericICollectionClass : ICollection<int>
+    class GenericIEnumerableClass : IEnumerable<int>
```

Comment:
Shouldn't this be reverted back to be an `ICollection<T>`? The test for `IEnumerable<T>` does not apply as we are not supporting those.

Reply:
The purpose of this class is for the failing tests. The fact used to use this class as evidence that the attribute throws an exception when applied on `ICollection<T>`. Now that it's been fixed, I purposely minimized this class to `IEnumerable<T>`.
I'd hope we support `IEnumerable<object>` with greedy iteration to the point we know the result as I pointed out [here](https://github.com/dotnet/corefx/issues/23461#issuecomment-324772257), for completeness, but I guess it's a no.

Reply:
Got it. Is there a positive test case for the added scenario?

Reply:
I added one as [requested](https://github.com/dotnet/corefx/pull/23664#pullrequestreview-59663873).
Will commit soon.